src/ApiClient.php

```diff
@@ -0,0 +1,210 @@
+
+        return $this->jsonMapper->mapArray($arr, 'BookingProtect\InsuranceHub\Client\Policy');
+    }
+
+    /**
+     * @throws InsureHubApiAuthorisationException
+     * @throws InsureHubApiAuthenticationException
+     * @throws InsureHubApiException
+     * @throws InsureHubApiValidationException
+     * @throws Exception
+     */
+    public function searchForPolicyByOfferingId(PolicySearchByOfferingId $policySearch): ?Policy {
+        $result = $this->execute($this->apiClientUrlBuilder->policySearchByOfferingIdUrl(), 'POST', $policySearch);
+
+        $data = $result->getBody()->getContents();
+        if (is_null($data)) {
+            return NULL;
+        }
+
+        return $this->jsonMapper->map(json_decode($data), new Policy());
+    }
+
+    /**
+     * @throws InsureHubApiAuthorisationException
+     * @throws InsureHubApiAuthenticationException
+     * @throws InsureHubApiException
+     * @throws InsureHubApiValidationException
+     * @throws Exception
+     */
+    public function getAdjustmentOffering(AdjustmentRequest $adjustmentRequest): AdjustmentOffering {
+        $result = $this->execute($this->apiClientUrlBuilder->adjustmentRequestUrl(), 'POST', $adjustmentRequest);
+
+        return $this->jsonMapper->map(json_decode($result->getBody()->getContents()), new AdjustmentOffering());
+    }
+
+    /**
+     * @throws InsureHubApiAuthorisationException
+     * @throws InsureHubApiAuthenticationException
+     * @throws InsureHubApiException
+     * @throws InsureHubApiValidationException
+     */
+    public function submitAdjustmentResult(AdjustmentOfferingResult $adjustmentResult): bool {
+        $result = $this->execute($this->apiClientUrlBuilder->adjustmentResultUrl(), 'POST', $adjustmentResult);
+
+        return $result->getStatusCode() == 201;
+    }
+
+    /**
+     * @throws InsureHubApiAuthorisationException
+     * @throws InsureHubApiAuthenticationException
+     * @throws InsureHubApiException
+     * @throws InsureHubApiValidationException
+     */
+    public function cancelSale(CancellationRequest $cancellationRequest): bool {
+        $result = $this->execute($this->apiClientUrlBuilder->cancellationUrl(), 'POST', $cancellationRequest);
+
+        return $result->getStatusCode() == 201;
+    }
+
+    /**
+     * @param string $url
+     * @param string $method
+     * @param ?JsonSerializable $requestBody
+     *
+     * @return ResponseInterface
+     *
+     * @throws InsureHubApiAuthenticationException
+     * @throws InsureHubApiAuthorisationException
+     * @throws InsureHubApiException
+     * @throws InsureHubApiValidationException
+     */
+    private function execute(string $url, string $method, ?JsonSerializable $requestBody = null): ResponseInterface {
+        $authToken = $this->authTokenGenerator->generateToken($this->configuration->vendorId, $this->configuration->apiKey);
+
+        $body = $requestBody ? json_encode($requestBody) : null;
+        $client = new Client();
```

Comment:
Could we also change this to be a dependency on the constructor rather than instantiate it here.  Again this will help with testability and dependency injection.  Looking at the documents for Guzzle it appears that they promote re-use of the client as this re-uses the curl handle - this also reduce the risk of socket exhaustion if you see a large spike in traffic.